src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/ByteSequenceComparer.cs

```diff
@@ -18,71 +18,17 @@ private ByteSequenceComparer()
 
         internal static bool Equals(ImmutableArray<byte> x, ImmutableArray<byte> y)
         {
-            if (x == y)
-            {
-                return true;
-            }
-
-            if (x.IsDefault || y.IsDefault || x.Length != y.Length)
-            {
-                return false;
-            }
-
-            for (var i = 0; i < x.Length; i++)
-            {
-                if (x[i] != y[i])
-                {
-                    return false;
-                }
-            }
-
-            return true;
+            return x.AsSpan().SequenceEqual(y.AsSpan());
         }
 
         internal static bool Equals(byte[] left, int leftStart, byte[] right, int rightStart, int length)
         {
-            if (left == null || right == null)
-            {
-                return ReferenceEquals(left, right);
-            }
-
-            if (ReferenceEquals(left, right) && leftStart == rightStart)
-            {
-                return true;
-            }
-
-            for (var i = 0; i < length; i++)
-            {
-                if (left[leftStart + i] != right[rightStart + i])
-                {
-                    return false;
-                }
-            }
-
-            return true;
+            return left.AsSpan(leftStart, length).SequenceEqual(right.AsSpan(rightStart, length));
         }
 
         internal static bool Equals(byte[]? left, byte[]? right)
         {
-            if (ReferenceEquals(left, right))
-            {
-                return true;
-            }
-
-            if (left == null || right == null || left.Length != right.Length)
-            {
-                return false;
-            }
-
-            for (var i = 0; i < left.Length; i++)
-            {
-                if (left[i] != right[i])
-                {
-                    return false;
-                }
-            }
-
-            return true;
+            return left.AsSpan().SequenceEqual(right.AsSpan());
```

Comment:
This isn't always the same answer as what was there before, e.g. if left is `new byte[0]` and right is null, previously this would have returned false but now it'll return true.  Does that affect any consumers?

Reply:
I don't think this method will ever get `null` arrays, it is only used by the dictionary that caches entries to the Blob Heap in `MetadataBuilder`, and its `GetOrAddBlob` methods check for `null`.

https://github.com/dotnet/runtime/blob/cd553b7208db0df889d8fbb325de65182443bb31/src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/MetadataBuilder.Heaps.cs#L206-L228

Reply:
Thanks. What about the other Equals being changed?  Similar possible issues if nulls are possible.